pkg/aws/client/client.go

```diff
@@ -1558,8 +1643,10 @@ func (c *Client) UpdateSubnetAttributes(ctx context.Context, desired, current *S
 		modified = true
 	}
 	privateDnsHostnameTypeOnLaunch := desired.PrivateDnsHostnameTypeOnLaunch
-	if privateDnsHostnameTypeOnLaunch == nil {
+	if privateDnsHostnameTypeOnLaunch == nil && desired.CidrBlock != "" && !(desired.Ipv6Native != nil && *desired.Ipv6Native) {
 		privateDnsHostnameTypeOnLaunch = aws.String(ec2.HostnameTypeIpName)
+	} else {
+		privateDnsHostnameTypeOnLaunch = aws.String(ec2.HostnameTypeResourceName)
```

Comment:
This should be identical to
```golang
	privateDnsHostnameTypeOnLaunch := aws.String(ec2.HostnameTypeResourceName)
	if desired.PrivateDnsHostnameTypeOnLaunch == nil && desired.CidrBlock != "" && !(desired.Ipv6Native != nil && *desired.Ipv6Native) {
		privateDnsHostnameTypeOnLaunch = aws.String(ec2.HostnameTypeIpName)
	}
```
which reads strange to me. If the desired DnsHostnameType is unset, we set it to IpName if IPv4 is enabled, I think. All other cases will just lead to using the resource name type.
Are there no good reasons to honour a desired value of `ec2.HostnameTypeIpName`, because that will not be possible with the change?

Reply:
Could catch! Fixed.